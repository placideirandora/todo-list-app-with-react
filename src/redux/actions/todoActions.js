// Action types
export const ADD_TODO = 'ADD_TODO';
export const REMOVE_TODO = 'REMOVE_TODO';

// Action creators
export const addTodo = content => {
  return {
    type: ADD_TODO,
    payload: content
  };
};

export const removeTodo = id => {
  return {
    type: REMOVE_TODO,
    payload: id
  };
};
