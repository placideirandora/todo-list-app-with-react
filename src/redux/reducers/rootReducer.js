const initialState = {
  todos: []
};

const rootReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ADD_TODO':
      action.id = Math.floor(Math.random() * 1000);

      return {
        todos: [...state.todos, action]
      };

    case 'REMOVE_TODO':
      const currentTodos = state.todos.filter(todo => {
        return action.payload !== todo.id;
      });

      return {
        ...state,
        todos: currentTodos
      };

    default:
      return state;
  }
};

export default rootReducer;
