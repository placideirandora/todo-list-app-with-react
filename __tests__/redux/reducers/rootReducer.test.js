import rootReducer from '../../../src/redux/reducers/rootReducer';
import { ADD_TODO, REMOVE_TODO } from '../../../src/redux/actions/todoActions';

describe('Root Reducer Test Cases', () => {
  it('should return the default state that is empty', () => {
    const action = { type: 'DUMMY_ACTION' };

    expect(rootReducer(undefined, action)).toMatchSnapshot();
  });

  it('should return the correct state while added a todo', () => {
    const action = { type: ADD_TODO, payload: 'Read Books', id: 1 };
    const expectedState = {
      todos: [{ type: ADD_TODO, payload: 'Read Books', id: 1 }]
    };

    expect(rootReducer(undefined, action)).toEqual(expectedState);
  });

  it('should return a defined state while added a todo without id', () => {
    const action = { type: ADD_TODO, payload: 'Make Video Games' };

    expect(rootReducer(undefined, action)).toBeDefined();
  });

  it('should return the correct state while removed a todo', () => {
    const state = {
      todos: [{ type: ADD_TODO, payload: 'Watch Movies', id: 1 }]
    };
    const action = { type: REMOVE_TODO, payload: 1 };
    const expectedState = {
      todos: []
    };

    expect(rootReducer(state, action)).toEqual(expectedState);
  });
});
