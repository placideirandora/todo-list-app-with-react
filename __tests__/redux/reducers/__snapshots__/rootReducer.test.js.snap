// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Reducer Test Cases should return the default state that is empty 1`] = `
Object {
  "todos": Array [],
}
`;
